### DISCUSSION TITLE
modify the response format?

### DISCUSSION THREAD

**Discussion by leoborlot** [docker-mailserver/docker-mailserver#3586] [General]
Hi guys. Any one know where the output template is set?
I want to try modify the output format to rescue and convert in JSON.

I want to modify this string format:

```
root@h01:/home/ubuntu# docker exec mailserver bash setup email list
* admin@email.com.br ( 9.0K / 15G ) [0%]
    [ aliases -> postmaster@email.com.br ]
```

**Comment by polarathene** [comment]
`setup email list` calls [this script](https://github.com/docker-mailserver/docker-mailserver/blob/811a7698459ebd40cbd520740c2074f4ad7402a6/target/bin/listmailuser).

This is the part that will iterate through each line of `postfix-accounts.cf` to output the list results:

https://github.com/docker-mailserver/docker-mailserver/blob/811a7698459ebd40cbd520740c2074f4ad7402a6/target/bin/listmailuser#L21-L26

Account, optional quota storage and aliases are then formatted here, with the final part concatenating into a single multi-line string per item:

https://github.com/docker-mailserver/docker-mailserver/blob/811a7698459ebd40cbd520740c2074f4ad7402a6/target/bin/listmailuser#L32-L43

Quota support queries Dovecot via the `doveadm` command, and parses that stdout response to get the data we're interested in:

https://github.com/docker-mailserver/docker-mailserver/blob/811a7698459ebd40cbd520740c2074f4ad7402a6/target/bin/listmailuser#L48-L77

Alias support is a bit tricky, using grep and sed commands to find aliases in `postfix-aliases.cf` for the account:

https://github.com/docker-mailserver/docker-mailserver/blob/811a7698459ebd40cbd520740c2074f4ad7402a6/target/bin/listmailuser#L79-L102

---

Alternatively, you could parse the output of `setup email list`?

If you do rewrite a JSON output, you may have some luck contributing it via PR as an alternative `setup email list --json`, but it must correctly list the data.

**Reply by leoborlot** [reply]
First now, I resolve with RE, break line in '*', and use RE to split the components, and solve. 
Next time I will try to build this job and create a more solid output. 
RE is good but, a . on the road break all. 

If any one want to try my soluttion in python, this is the code.  (the variables was in pt-br, next time bring more international)

 ```
   try:
        command = 'bash setup email list'
        result = execute_docker_command(command)
        partes = result.split('* ')
        padrao = r'(\S+) \(\s*([\d\.]+[KMG]?|~) / ([\d\.]+[KMG]?|~) \) \[([\d\.]+)%\](?:\s+\[ aliases -> (\S+(?:, \S+)*) \])?'
        dados = []

        for parte in partes[1:]:  # Ignorar a primeira parte vazia após a divisão
            correspondencia = re.findall(padrao, parte, re.M)
            if correspondencia:
                email, usado, total, porcentagem, aliases = correspondencia[0]
                aliases = {'aliases': aliases.split(', ')} if aliases else {}

                # Crie um dicionário com os dados extraídos
                item = {
                    'dominio': re.search(r'@(.+)', email).group(1) if email else None,
                    'email': email,
                    'usado': usado,
                    'total': total,
                    'porcentagem': porcentagem,
                    **aliases  # Adicione o dicionário de aliases, se existir
                }
                dados.append(item)
                
```

This is the output in JSON

<img width="519" alt="image" src="https://github.com/docker-mailserver/docker-mailserver/assets/20361821/0159969b-7be4-49a1-894a-a269a6f16838">
